diff --git a/lib/commonjs/lib/extract/extractProps.js b/lib/commonjs/lib/extract/extractProps.js
index 618a08fd1b48a9bfe87cc9664ce5a1e8c87f0b3b..bc9f64ebe0a16c68bacb4cf72d22d726474d6c3f 100644
--- a/lib/commonjs/lib/extract/extractProps.js
+++ b/lib/commonjs/lib/extract/extractProps.js
@@ -105,7 +105,7 @@ function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(_util.idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/commonjs/lib/extract/extractProps.windows.js b/lib/commonjs/lib/extract/extractProps.windows.js
index 7e602b5c808e246ddf6f02d0976f48bc3236ea59..03704b2e07fff6dea6469ecda4fb637ae6297201 100644
--- a/lib/commonjs/lib/extract/extractProps.windows.js
+++ b/lib/commonjs/lib/extract/extractProps.windows.js
@@ -93,7 +93,7 @@ function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(_util.idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/module/lib/extract/extractProps.js b/lib/module/lib/extract/extractProps.js
index 968888919514bfdc638745837b5afc32c7f20f41..69ed95f26b99c9af2347235bf0827164462716df 100644
--- a/lib/module/lib/extract/extractProps.js
+++ b/lib/module/lib/extract/extractProps.js
@@ -94,7 +94,7 @@ export default function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/module/lib/extract/extractProps.windows.js b/lib/module/lib/extract/extractProps.windows.js
index 4f98597b1f6f6c309ed5fc6e74b6349eb9626246..d39443b48f845d0882510c452c96fdff889fea80 100644
--- a/lib/module/lib/extract/extractProps.windows.js
+++ b/lib/module/lib/extract/extractProps.windows.js
@@ -82,7 +82,7 @@ export default function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/src/lib/extract/extractProps.ts b/src/lib/extract/extractProps.ts
index 1f30ce0c62fe2d1775f3acbf8b8b5a08dc52f5e0..2d13671fd7fcc3a08a25440319cec5a9656e5283 100644
--- a/src/lib/extract/extractProps.ts
+++ b/src/lib/extract/extractProps.ts
@@ -145,7 +145,7 @@ export default function extractProps(
     }
   }
 
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(idPattern);
 
     if (matched) {
diff --git a/src/lib/extract/extractProps.windows.ts b/src/lib/extract/extractProps.windows.ts
index 14c783b56866b1cc2b986b202f5b536d0cd3175d..44cd320502e676ed1ab45aa8bcab6eb2c6bd9c25 100644
--- a/src/lib/extract/extractProps.windows.ts
+++ b/src/lib/extract/extractProps.windows.ts
@@ -127,7 +127,7 @@ export default function extractProps(
     }
   }
 
-  if (mask) {
+  if (typeof mask === "string") {
     const matched = mask.match(idPattern);
 
     if (matched) {