diff --git a/lib/commonjs/lib/extract/extractProps.js b/lib/commonjs/lib/extract/extractProps.js
index 618a08fd1b48a9bfe87cc9664ce5a1e8c87f0b3b..78900d4ef38598209a2d7e433928918452dc4cc6 100644
--- a/lib/commonjs/lib/extract/extractProps.js
+++ b/lib/commonjs/lib/extract/extractProps.js
@@ -105,7 +105,7 @@ function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(_util.idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/commonjs/lib/extract/extractProps.windows.js b/lib/commonjs/lib/extract/extractProps.windows.js
index 7e602b5c808e246ddf6f02d0976f48bc3236ea59..69d6a2f205b6fc2f3d00a280a61383973f4fa5f2 100644
--- a/lib/commonjs/lib/extract/extractProps.windows.js
+++ b/lib/commonjs/lib/extract/extractProps.windows.js
@@ -93,7 +93,7 @@ function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(_util.idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/module/lib/extract/extractProps.js b/lib/module/lib/extract/extractProps.js
index 968888919514bfdc638745837b5afc32c7f20f41..1edcd4b36604b2efe3dac5360c091c1e8c9b2099 100644
--- a/lib/module/lib/extract/extractProps.js
+++ b/lib/module/lib/extract/extractProps.js
@@ -94,7 +94,7 @@ export default function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/lib/module/lib/extract/extractProps.windows.js b/lib/module/lib/extract/extractProps.windows.js
index 4f98597b1f6f6c309ed5fc6e74b6349eb9626246..aba80c76a7ad79e6f911d064a6c465341d24fa40 100644
--- a/lib/module/lib/extract/extractProps.windows.js
+++ b/lib/module/lib/extract/extractProps.windows.js
@@ -82,7 +82,7 @@ export default function extractProps(props, ref) {
       console.warn('Invalid `clipPath` prop, expected a clipPath like "#id", but got: "' + clipPath + '"');
     }
   }
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(idPattern);
     if (matched) {
       extracted.mask = matched[1];
diff --git a/src/lib/extract/extractProps.ts b/src/lib/extract/extractProps.ts
index 378aaf998ac03caf412b293f5088376e050d911a..a92fa82ac27bb202f9f9ea71b55018d41d15028c 100644
--- a/src/lib/extract/extractProps.ts
+++ b/src/lib/extract/extractProps.ts
@@ -145,7 +145,7 @@ export default function extractProps(
     }
   }
 
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(idPattern);
 
     if (matched) {
diff --git a/src/lib/extract/extractProps.windows.ts b/src/lib/extract/extractProps.windows.ts
index c2200a0aeb197a36baea6b87c84c45b61dfbda3a..d213694a93aaad33b2a9f711ff9b389946934f67 100644
--- a/src/lib/extract/extractProps.windows.ts
+++ b/src/lib/extract/extractProps.windows.ts
@@ -127,7 +127,7 @@ export default function extractProps(
     }
   }
 
-  if (mask) {
+  if (mask && typeof mask === 'string') {
     const matched = mask.match(idPattern);
 
     if (matched) {
